### PR TITLE
Telemetry + Bug Fixes from performance branch

### PR DESCRIPTION
**Pull request by ejfitzgerald**
In addition this PR also disables the `DebugMutex` by default. It can be reenabled via CMake option if you really want it switched on 😁

### CODE REVIEW

File: libs/ledger/src/chain/block_coordinator.cpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ using DAGPtr               = std::shared_ptr<ledger::DAGInterface>;
 
 // Constants
 const std::chrono::milliseconds TX_SYNC_NOTIFY_INTERVAL{1000};
-const std::chrono::milliseconds EXEC_NOTIFY_INTERVAL{500};
+const std::chrono::milliseconds EXEC_NOTIFY_INTERVAL{5000};
```
Comment (ejfitzgerald) [inline libs/ledger/src/chain/block_coordinator.cpp:61]: This was just too short

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,19 +9,20 @@ project(fetch)
...
-option(FETCH_DISABLE_COLOUR_LOG     "Disable the colour logging"        OFF)
-option(FETCH_STATIC_LINK            "Enable static linking"             OFF)
+option(FETCH_ENABLE_TESTS               "Enable test targets"                   ON)
+option(FETCH_ENABLE_BENCHMARKS          "Enable benchmarking targets"           ON)
+option(FETCH_ENABLE_EXAMPLES            "Enable example targets"                ON)
+option(FETCH_ENABLE_BENCHMARKS          "Enable benchmark targets"              ON)
+option(FETCH_ENABLE_PYTHON_BINDINGS     "Build Python examples"                 ON)
+option(FETCH_ENABLE_CCACHE              "Speed build times with ccache"         ON)
+option(FETCH_WARNINGS_AS_ERRORS         "Enable warnings as errors"             ON)
+option(FETCH_ENABLED_DEADLOCK_DETECTION "Enable deadlock detection on mutexes"  OFF)
```
Comment (wkaluza) [inline CMakeLists.txt]: For consistency
```suggestion
option(FETCH_ENABLE_DEADLOCK_DETECTION "Enable deadlock detection on mutexes"  OFF)
```

File: cmake/BuildTargets.cmake [hunk 3]
```diff
@@ -143,6 +143,11 @@ macro (setup_compiler)
     set(FETCH_COMPILE_LOGGING_LEVEL "info")
   endif ()
 
+  # debug mutex configuration
+  if (FETCH_ENABLED_DEADLOCK_DETECTION)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFETCH_DEBUG_MUTEX")
+  endif (FETCH_ENABLED_DEADLOCK_DETECTION)
```
Comment (wkaluza) [inline cmake/BuildTargets.cmake]: FETCH_ENABLED_DEADLOCK_DETECTION -> FETCH_ENABLE_DEADLOCK_DETECTION

File: apps/constellation/manifest_builder.cpp [hunk 4]
```diff
@@ -64,7 +64,7 @@ void GenerateDefaultManifest(std::string const &external_address, uint16_t port,
   // register all of the lanes (storage shards)
   for (uint32_t i = 0; i < num_lanes; ++i)
   {
-    peer.Update(external_address, static_cast<uint16_t>(port + STORAGE_PORT_OFFSET + i));
+    peer.Update(external_address, static_cast<uint16_t>(port + STORAGE_PORT_OFFSET + (2 * i)));
```
Comment (wkaluza) [inline apps/constellation/manifest_builder.cpp:67]: I don't think I understand this change. Can you let me know f2f what's going on here?

File: CMakeLists.txt [hunk 5]
```diff
@@ -9,19 +9,20 @@ project(fetch)
...
-option(FETCH_ENABLE_CCACHE          "Speed build times with ccache"     ON)
-option(FETCH_WARNINGS_AS_ERRORS     "Enable warnings as errors"         ON)
-option(FETCH_VERBOSE_CMAKE          "Enable verbose cmake output"       OFF)
-option(FETCH_ENABLE_CLANG_TIDY      "Enable clang tidy checking"        OFF)
-option(FETCH_ENABLE_COVERAGE        "Enable coverage generation"        OFF)
-option(FETCH_ENABLE_METRICS         "Enable the collection of metrics"  OFF)
-option(FETCH_DISABLE_COLOUR_LOG     "Disable the colour logging"        OFF)
-option(FETCH_STATIC_LINK            "Enable static linking"             OFF)
+option(FETCH_ENABLE_TESTS               "Enable test targets"                   ON)
+option(FETCH_ENABLE_BENCHMARKS          "Enable benchmarking targets"           ON)
```
Comment (wkaluza) [inline CMakeLists.txt:13]: Duplicate option
```suggestion
```

**Review by wkaluza — CHANGES_REQUESTED** [review]
Niggles only

File: CMakeLists.txt (hunk 2 above)
Comment (ejfitzgerald) [reply CMakeLists.txt]: Nope good spot

File: apps/constellation/manifest_builder.cpp (hunk 4 above)
Comment (ejfitzgerald) [reply apps/constellation/manifest_builder.cpp:67]: Just an inconsistency from an older piece of the logic. Each lane shard has 2 network inferfaces so this is represented here.

File: CMakeLists.txt (hunk 5 above)
Comment (ejfitzgerald) [reply CMakeLists.txt:13]: Curious, this is in master actually. Seems to be a previous merge issue

Comment (wkaluza) [reply CMakeLists.txt:13]: Yes, I understand you did not introduce it here. :)

**Review by wkaluza — APPROVED** [review]

**Pull request merged**